.github/workflows/run-regression-tests.yaml

```diff
@@ -88,6 +88,25 @@ env:
   REG_TESTS_GH_REPO: 4chain-ag/spv-wallet-regression
 
 jobs:
+  check-balance:
```

Comment:
AFAIU all the other jobs should only be started if the balance is ok.
In other words the `preparation` job should have `needs: check-balance`, so in case of failure of the check balance the preparation won't be started and the rest (hopefully) also.